r/src/array.cpp

```diff
@@ -539,7 +539,7 @@ inline SEXP StringArray_to_Vector(const std::shared_ptr<arrow::Array>& array) {
 
   Rcpp::CharacterVector res(no_init(n));
   auto p_offset = GetValuesSafely<int32_t>(array->data(), 1, array->offset());
-  auto p_data = GetValuesSafely<char>(array->data(), 2, *p_offset);
+  auto p_data = GetValuesOrNull<char>(array->data(), 2, *p_offset);
```

Comment:
I'm not sure what the purpose of `GetValuesSafely` is now

Reply:
I think the correct behavior is always to return null (for length-0 arrays), not to have an error

Reply:
Makes sense, changed. Ran the tests, so far so good; but would definitely wait for @romainfrancois feedback before merging.